src/Angor/Avalonia/Angor.Wallet/Infrastructure/Impl/TransactionWatcher.cs

```diff
@@ -0,0 +1,35 @@
+using System.Reactive.Linq;
+using Angor.Wallet.Application;
+using Angor.Wallet.Domain;
+using CSharpFunctionalExtensions;
+
+namespace Angor.Wallet.Infrastructure.Impl
+{
+    public class TransactionWatcher(IWalletAppService walletAppService) : ITransactionWatcher
+    {
+        public IObservable<Result<BroadcastedTransaction>> Watch(WalletId id)
+        {
+            var initial = Observable.Defer(() => GetTransactions(id));
+            
+            var polling = Observable.Timer(TimeSpan.FromSeconds(10))
```

Comment:
There is no point calling for new transactions if we are on the same block, this is also a lot of work in the wallet app service, we could check what is the block hash of the tip and check if it is different to ours before calling getTransactions so it would only refresh every 10 minutes (or 1 minute in signet etc.)
But that might be an overkill as well, perhaps there should be a trigger on the UI like when I make an investment etc. to start pulling until we get the expected change - otherwise I think the user can just refresh manually if something is missing.
What do you think?

Reply:
We can implement any trigger later to avoid unnecessary call to the indexer. Your suggestion sounds great, by the way!